ext/ffi/lib.rs

```diff
@@ -1031,40 +1027,37 @@ fn ffi_parse_pointer_arg(
       })?
       .get_backing_store();
     if byte_offset > 0 {
-      &backing_store[byte_offset..] as *const _ as *const u8
+      &backing_store[byte_offset..] as *const _ as *const c_void
     } else {
-      &backing_store[..] as *const _ as *const u8
+      &backing_store[..] as *const _ as *const c_void
     }
-  } else if let Ok(value) = v8::Local::<v8::BigInt>::try_from(arg) {
-    value.u64_value().0 as usize as *const u8
-  } else if let Ok(value) = v8::Local::<v8::Number>::try_from(arg) {
-    value.integer_value(scope).unwrap() as usize as *const u8
+  } else if let Ok(value) = v8::Local::<v8::External>::try_from(arg) {
+    value.value() as *const c_void
   } else if let Ok(value) = v8::Local::<v8::ArrayBuffer>::try_from(arg) {
     let backing_store = value.get_backing_store();
-    &backing_store[..] as *const _ as *const u8
+    &backing_store[..] as *const _ as *const c_void
   } else if arg.is_null() {
     ptr::null()
   } else {
     return Err(type_error("Invalid FFI pointer type, expected null, integer, BigInt, ArrayBuffer, or ArrayBufferView"));
   };
+  let pointer = pointer as *mut c_void;
   Ok(NativeValue { pointer })
 }
 
 #[inline]
 fn ffi_parse_function_arg(
-  scope: &mut v8::HandleScope,
+  _scope: &mut v8::HandleScope,
   arg: v8::Local<v8::Value>,
 ) -> Result<NativeValue, AnyError> {
   // Order of checking:
```

Comment:
This comment is now out of sync with the code.

Reply:
Will fix if we move forward with this PR.